src/Session/Session.cc

```diff
@@ -362,7 +368,7 @@ bool Session::FillFileInfo(
     // Resolve filename and fill file info submessage
     bool file_info_ok(false);
 
-    fullname = GetResolvedFilename(_top_level_folder, folder, filename, message);
+    fullname = GetResolvedFilename(ProgramSettings::GetInstance().top_level_folder, folder, filename, message);
```

Comment:
`Session` uses the settings in multiple places. To avoid repeating this verbose syntax, I would suggest adding a `ProgramSettings& _settings` to this class, and setting it to the instance in the constructor. It will then be possible to refer to e.g. `_settings.top_level_folder`.